### PR TITLE
Add openai instrumentation to opentelemetry-bootstrap

### PR DESCRIPTION
**Pull request by lzchen**
Fixes https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2995

Note this does NOT include openai instrumentation as a default installation but will install it if user has `openai` installed.

### CODE REVIEW

File: scripts/otel_packaging.py [hunk 1]
```diff
@@ -21,14 +21,24 @@
...
 def get_instrumentation_packages():
-    for pkg in sorted(os.listdir(instrumentations_path)):
+    pkg_paths = []
+    for pkg in os.listdir(instrumentations_path):
         pkg_path = os.path.join(instrumentations_path, pkg)
         if not os.path.isdir(pkg_path):
             continue
+        pkg_paths.append(pkg_path)
+    for pkg in os.listdir(genai_instrumentations_path):
+        pkg_path = os.path.join(genai_instrumentations_path, pkg)
```
Comment (karthikscale3) [inline scripts/otel_packaging.py:35]: nice, I guess this logic bootstraps the necessary packages?

**Review by karthikscale3 — APPROVED** [review]

File: scripts/otel_packaging.py (hunk 1 above)
Comment (lzchen) [reply scripts/otel_packaging.py:35]: @karthikscale3 

Yes! It scans the packages a user has that can be instrumented and installs the relevant instrumentation if exists. Usually the first step before running zero-code mechanisms like `opentelemetry-instrument`.

**Review by nirga — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
Thanks, so this adds the dep "opentelemetry-instrumentation-openai-v2", which users could also do manually if they don't use opentelemetry-bootstrap to setup the project. Sounds great!

**Comment by lzchen** [comment]
@codefromthecrypt  yup! It all helps with the zero-code story.

**Pull request merged**